compiler-core/src/bit_string.rs

```diff
@@ -234,15 +234,7 @@ fn type_options<T>(
         return err(ErrorType::UnitMustHaveSize, unit.location());
     }
 
-    // size cannot be used with float
-    match categories {
-        SegmentOptionCategories {
-            typ: Some(Float { .. }),
-            size: Some(opt),
-            ..
-        } => err(ErrorType::FloatWithSize, opt.location()),
-        _ => Ok(categories.segment_type()),
-    }
+    Ok(categories.segment_type())
```

Comment:
Is FloatWithSize still used?

Reply:
should be used in the future to validate static size of floats to 16/32/64 only, not used now

Reply:
actually will add that now(FloatWithSize), lacked enough understanding to add it two days ago, will also add the runtime tests

Reply:
Nice, thanks!